knowledge-base/treeview-drag-to-custom-target.md

```diff
@@ -0,0 +1,206 @@
+---
+title: Drag TreeView Items to a Custom Target
+description: How to support draging and reordering items inside the TreeView but also allow dragging and dropping items outside the component to a custom target
+type: how-to
+page_title: How to Drag TreeView Items to a Custom Target
+slug: treeview-kb-drag-to-custom-target
+position: 
+tags: 
+ticketid: 1592132
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>TreeView for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+I have a TreeView that allows dragging and reordering items inside the component. I want to also allow dragging items outside the TreeView, so the users can drop them on custom target. How to achieve that?
+
+## Solution
+
+To allow dragging outside the TreeView and dropping on a custom target, follow these steps:
+
+1. Handle the [`OnDragStart`]({%slug treeview-events%}#drag-events) event of the TreeView to get the item that the user started dragging.
+1. Handle the [`@onpointerup`](https://developer.mozilla.org/en-US/docs/Web/API/Element/pointerup_event) event of your custom target to add the TreeView inside as needed. Based on product specifics, the TreeView internally does not use draggable events, it uses pointer events to handle the dragging. Therefore, `@ondrop` of your custom target will not fire when dragging a TreeView item in it as the component expects a subsequent pointer event. Use the `@onpointerup` instead of `@ondrop`.
```

Comment:
Ideally, [external links should open in a new window](https://docs.telerik.com/style-guide/cross-references#guidelines) with the `<a href="" target="_blank">` syntax.